src/server/main_service.cc

```diff
@@ -589,6 +589,47 @@ void Service::Shutdown() {
   ThisFiber::SleepFor(10ms);
 }
 
+bool Service::CheckKeysOwnership(const CommandId* cid, CmdArgList args,
+                                 ConnectionContext* dfly_cntx) {
+  if (cid->first_key_pos() == 0) {
+    return true;  // No key command.
+  }
+  OpResult<KeyIndex> key_index_res = DetermineKeys(cid, args);
+  if (!key_index_res) {
+    (*dfly_cntx)->SendError(key_index_res.status());
+    return false;
+  }
+
+  const auto& key_index = *key_index_res;
+  SlotId keys_slot;
```

Comment:
@adiholden 

I got this warning on compilation and was about to ignore it. 
```
../src/server/main_service.cc: In member function ‘bool dfly::Service::CheckKeysOwnership(const dfly::CommandId*, facade::CmdArgList, dfly::ConnectionContext*)’:
../src/server/main_service.cc:625:49: warning: ‘keys_slot’ may be used uninitialized in this function [-Wmaybe-uninitialized]
  625 |   if (!server_family_.cluster_config()->IsMySlot(keys_slot)) {
```

But it seems that slot_id can actually be non initialized. Because even if `cid->first_key_pos() > 0`, the command itself can have 0 keys if it uses variadic keys. For example: `EVAL {script} 0`.

Reply:
thanks @dranikpg